src/Microsoft.Health.Fhir.Core/Features/Search/Registry/SearchParameterStatusManager.cs

```diff
@@ -191,16 +191,25 @@ internal async Task ApplySearchParameterStatus(IReadOnlyCollection<ResourceSearc
 
             foreach (var paramStatus in updatedSearchParameterStatus)
             {
-                var param = _searchParameterDefinitionManager.GetSearchParameter(paramStatus.Uri);
-
-                var tempStatus = EvaluateSearchParamStatus(paramStatus);
+                if (_searchParameterDefinitionManager.TryGetSearchParameter(paramStatus.Uri, out var param))
+                {
+                    var tempStatus = EvaluateSearchParamStatus(paramStatus);
 
-                param.IsSearchable = tempStatus.IsSearchable;
-                param.IsSupported = tempStatus.IsSupported;
-                param.IsPartiallySupported = tempStatus.IsPartiallySupported;
-                param.SortStatus = paramStatus.SortStatus;
+                    param.IsSearchable = tempStatus.IsSearchable;
+                    param.IsSupported = tempStatus.IsSupported;
+                    param.IsPartiallySupported = tempStatus.IsPartiallySupported;
+                    param.SortStatus = paramStatus.SortStatus;
 
-                updated.Add(param);
+                    updated.Add(param);
+                }
+                else if (!updatedSearchParameterStatus.Where(p => p.Uri.Equals(paramStatus) && p.Status == SearchParameterStatus.Deleted).Any())
```

Comment:
>paramStatus [](http://example.com/codeflow?start=79&length=11)

shouldn't this be paramStatus.Uri?

Reply:
Nit: I think this could also be simplified to

`else if (!updatedSearchParameterStatus.Any(p => p.Uri.Equals(paramStatus.Uri) && p.Status == SearchParameterStatus.Deleted))`

Reply:
Thanks!  Good catch.